lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1451,35 +1455,46 @@ mrs_free(void *ptr)
 void *
 mrs_mallocx(size_t size, int flags)
 {
+	size_t align = MALLOCX_ALIGN_GET(flags);
+	void *ret;
+
 	if (!quarantining)
 		return (REAL(mallocx)(size, flags));
 
-	/*
-	 * This API supports optional zeroing.  Since no one really uses
-	 * it, we don't care about performance and preserve the ABI by
-	 * unconditionally calling calloc.  (We should always be zeroing
-	 * for correctness anyway.)
-	 */
-	return (mrs_calloc(1, size));
+	if (align <= CAPREVOKE_BITMAP_ALIGNMENT)
+		ret = mrs_malloc(size);
+	else if (mrs_posix_memalign(&ret, size, align) != 0)
+		ret = NULL;
+
+#ifndef CLEAR_ON_ALLOC
+	/* Clear if requested and we aren't clearing above. */
+	if (ret != NULL & (flags & MALLOCX_ZERO) != 0)
```

Comment:
```suggestion
	if (ret != NULL && (flags & MALLOCX_ZERO) != 0)
```
(though technically not a bug, as != does have higher precedence than &, famously, due to not wanting to patch a few hundred KiB across 3 machines to make the grammar more sensible)

Reply:
sigh